website/src/views/mpe/form/ModulesSelectContainer.tsx

```diff
@@ -0,0 +1,77 @@
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { ModuleSelectList, ModuleSelectListItem } from 'types/reducers';
+import { ModuleCode, Semester } from 'types/modules';
+import type { MpePreference } from 'types/mpe';
+
+import Online from 'views/components/Online';
+import { createSearchPredicate, sortModules } from 'utils/moduleSearch';
+import { State as StoreState } from 'types/state';
+import ModulesSelect from './ModulesSelect';
+
+type OwnProps = {
+  semester: Semester;
+  moduleList: ModuleSelectList;
+};
+
+type Props = OwnProps & {
+  preferences: MpePreference[];
+  addModule: (moduleCode: ModuleCode) => void;
+  removeModule: (moduleCode: ModuleCode) => Promise<void>;
+};
+
+const RESULTS_LIMIT = 500;
+
+function isModuleinPreference(preferences: MpePreference[], moduleCode: ModuleCode) {
+  return preferences.reduce<boolean>((acc, curr) => acc || curr.moduleCode === moduleCode, false);
+}
+function makeModuleList(state: StoreState, props: Props): ModuleSelectListItem[] {
+  return state.moduleBank.moduleList.map((module) => ({
+    ...module,
+    isAdded: isModuleinPreference(props.preferences, module.moduleCode),
```

Comment:
Avoid using linear search for membership test, since linear search is O(n) while set membership test is just O(1) 

```suggestion
function makeModuleList(state: StoreState, props: Props): ModuleSelectListItem[] {
  const selectedMpeModules = new Set(props.preferences.map(p => p.moduleCode));
  return state.moduleBank.moduleList.map((module) => ({
    ...module,
    isAdded: selectedMpeModules.has(module.moduleCode),
```

Reply:
Since preferences' length is 7, changing this to a set to complicate code is perhaps unnecessary. Will review in the future.